### DISCUSSION TITLE
Branch prediction

### DISCUSSION THREAD

**Discussion by iTheodoros** [mortbopet/Ripes#230] [Q&A] [answered]
Is there any branch prediction implemented on Ripes?

**Comment by mortbopet** [comment, marked as answer]
Nope! But i've always been interested in having it. It's something which could be implemented _somewhat_ agnostically of which processor model is currently loaded (much like how cache simulation is done). So if you'd like to try and implement branch prediction support, the cache work would probably be a good source of reference.

**Reply by iTheodoros** [reply]
Yes, that's what I plan to do as part of my senior design project.

**Reply by mortbopet** [reply]
Awesome! Feel free to reach out if you'd like my input as you go about this.
